website/client/src/components/shops/buyModal.vue

```diff
@@ -542,6 +542,9 @@ export default {
         }, 0);
         if (this.item.pinType === 'premiumHatchingPotion') {
           petsRemaining -= this.user.items.hatchingPotions[this.item.key] + 2 || 2;
+          if (this.item.path.indexOf('wackyHatchingPotions.') !== -1) {
+            petsRemaining -= 9;
```

Comment:
Instead of removing 9 from the count regardless of how many the user is trying to buy, shouldn't the count start from 9 in the case of wacky potions at line 534?

Reply:
That would be another option, but in that case we should also set it to 18 for normal premium hatching potions, 
instead of subtracting 2 extra.

Reply:
I was slightly confused by how this code was originally meant to work but I tried a couple of different combinations and it works well (20 - 2 - 9 leaves exactly 9 available to buy)

I think we should refactor it to something clearer but for the scope of this PR this looks good! I'll merge it at the end of the week